vcd/resource_vcd_org_vdc.go

```diff
@@ -247,7 +250,19 @@ func resourceVcdVdcCreate(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error waiting for vdc to finish: %#v", err)
 	}
 
-	d.SetId(d.Get("name").(string))
+	err = adminOrg.Refresh()
+	if err != nil {
+		log.Printf("[DEBUG] Unable to refresh org.")
+		return fmt.Errorf("unable to refresh org. %#v", err)
+	}
+
+	adminVdc, err := adminOrg.GetAdminVdcByName(d.Get("name").(string))
+	if err != nil || adminVdc == (govcd.AdminVdc{}) {
+		log.Printf("[DEBUG] Unable to find vdc.")
+		return fmt.Errorf("unable to find vdc.. %#v", err)
+	}
+
+	d.SetId(adminVdc.AdminVdc.ID)
```

Comment:
A need change, but just a safe check - do we have any place where people could be refering to this resource's `.id` field? This could break things for them

Reply:
Yes could brake, I can add in documentation explicitly saying about this. Though recommend to leave this - as this less chances of brakes now when later.

Reply:
True. Ok with me.

Reply:
Just thought - maybe worth adding some note to changelog near the IMPROVEMENTS.

Reply:
Added